src/main/js/helpers/legend.js

```diff
@@ -392,19 +394,36 @@ const loadPieLegendItem = (legendSVG, dataTarget, { hoverHandler }, config) => {
 const getDefaultLegendOptions = (graphConfig, dataTarget) => {
     const legendOptions = getDefaultValue(dataTarget.legendOptions, {
         showShape: true,
-        showLine: false,
+        showLine: dataTarget.showShapes === false,
         showElement: true
     });
-    legendOptions.showShape = getDefaultValue(legendOptions.showShape, true);
+    legendOptions.showShape =
+        dataTarget.showShapes !== false
+            ? getDefaultValue(legendOptions.showShape, true)
+            : false;
     legendOptions.showLine = getDefaultValue(legendOptions.showLine, false);
     legendOptions.showElement = getDefaultValue(
         legendOptions.showElement,
         true
     );
     legendOptions.style = getDefaultValue(legendOptions.style, {});
-    legendOptions.style = {
-        strokeDashArray: getStrokeDashArray(legendOptions.style)
-    };
+    if (legendOptions.style.strokeDashArray) {
+        legendOptions.style = {
+            strokeDashArray: getStrokeDashArray(legendOptions.style)
+        };
+    } else if (
+        dataTarget.showShapes === false &&
+        legendOptions.style.strokeDashArray === undefined
+    ) {
+        legendOptions.style.strokeDashArray = dataTarget.style.strokeDashArray;
+    } else if (
+        dataTarget.showShapes === true &&
+        legendOptions.style.strokeDashArray === undefined
+    ) {
+        legendOptions.style = {
+            strokeDashArray: getStrokeDashArray(legendOptions.style)
+        };
+    }
```

Comment:
Since we also need check for undefined(legendOptions.style.strokeDashArray === undefined), I don't think we can do it like this. Correct me if i am wrong.

Reply:
It covers that scenario as well in my code 

- if `legendOptions.style.strokeDashArray` defined, it will go into if block

- if `legendOptions.style.strokeDashArray` undefined, it will go into else block. So we don't have to explicitly check the value is undefined

Reply:
[Fixed](https://github.com/cerner/carbon-graphs/pull/297/commits/3216e7b3c43659582bdd66ad822d0b715eb16eb5#diff-fce2adcfc7080dc27fec3e9711017bacR416).